shopinvader_api_cart/tests/test_shopinvader_api_cart.py

```diff
@@ -0,0 +1,394 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import json
+
+from extendable import context
+from fastapi import status
+from requests import Response
+
+from odoo.exceptions import MissingError
+from odoo.tests.common import tagged
+
+from odoo.addons.extendable.registry import _extendable_registries_database
+from odoo.addons.fastapi.tests.common import FastAPITransactionCase
+
+from ..routers.cart import cart_router
+from ..schemas import CartTransaction
+
+# TODO: a second test class with a user that has no rights
+
+
+@tagged("post_install", "-at_install")
+class TestSaleCart(FastAPITransactionCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        extendable_registry = _extendable_registries_database.get(cls.env.cr.dbname)
+        cls.token = context.extendable_registry.set(extendable_registry)
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        cls.user_no_rights = cls.env["res.users"].create(
+            {
+                "name": "Test User Without Rights",
+                "login": "user_no_rights",
+            }
+        )
+        user_with_rights = cls.env["res.users"].create(
+            {
+                "name": "Test User With Rights",
+                "login": "user_with_rights",
+            }
+        )
+        user_with_rights.groups_id = [
+            (4, cls.env.ref("shopinvader_api_cart.sale_cart_user_group").id)
```

Comment:
I think that you should only give the group "sale_cart_user_group" to your user. Otherwise, it will have other groups due to the fact that it is an "internal user".

Shouldn't you do something like:

```
user_with_rights.groups_id = [
            (6,0, [cls.env.ref("shopinvader_api_cart.sale_cart_user_group").id])
        ]
```

Reply:
Done in the last commit, thanks